### ISSUE TITLE
No IP address on nebula1 interface

### ISSUE THREAD

**Issue by andrewmgee** [slackhq/nebula#591]
Have a bunch of nebula clients successfully working, but for some reason one of them won't assign the IP to the nebula1 interface when the service starts up. I have to manually run `ip addr add 10.10.10.10/24 dev nebula1` command, then it works. 

Only difference is I did a `systemctl disable nebula` for testing, then re-enabled. But don't' see why that would make a difference. Was working before that, when it was first installed.

Running on Ubuntu 20.04 LTS, on Pi 4

**ip a command after service start**
```
nebula1: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1300 qdisc fq_codel state UNKNOWN group default qlen 500
    link/none
    inet6 fe80::20fd:b88:381f:abdf/64 scope link stable-privacy
       valid_lft forever preferred_lft forever

```

**ip a command after manually running ip addr**
```
nebula1: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1300 qdisc fq_codel state UNKNOWN group default qlen 500
    link/none
    inet 10.10.10.10/24 scope global nebula1
       valid_lft forever preferred_lft forever
    inet6 fe80::20fd:b88:381f:abdf/64 scope link stable-privacy
       valid_lft forever preferred_lft forever

```

**systemctl status nebula**
```
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="UDP hole punching enabled"
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="Nebula interface is active" build=1.4.0 interface=nebula1 network=10.10.10.10/24 udpAddr="0.0.0.0:42858">
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="Handshake message sent" handshake="map[stage:1 style:ix_psk0]" initi...redacted>
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="Handshake message sent" handshake="map[stage:1 style:ix_psk0]" initi...redacted>
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="Handshake message sent" handshake="map[stage:1 style:ix_psk0]" initi...redacted>
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="Handshake message sent" handshake="map[stage:1 style:ix_psk0]" initi...redacted>
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="Handshake message received" certName=lighthouse1...redacted>
nebula[724137]: time="2021-11-20T11:36:33-08:00" level=info msg="Handshake message received" certName=lighthouse3...redacted>
```

**Comment by tarrenj** [comment]
Hmm, can you post a sanitized version of the `config.yml` for that host please?

Additionally, are you sure you specified the IP correctly when creating the cert? `./nebula-cert print -path /path/to/cert.crt`

**Comment by andrewmgee** [comment]
@tarrenj see below for sanitized config.yml. Yes the correct IP was specified when creating the cert, it does work/connect after I manually do the `ip addr` command. Ie. I can ping across to other nebula IP's.

```
pki:
  ca: /etc/nebula/ca.crt
  cert: /etc/nebula/cert.crt
  key: /etc/nebula/key.key

static_host_map:
  "11.11.11.11": ["111.111.111.111:4001"]
  "22.22.22.22": ["222.222.222.222:4002"]
  "33.33.33.33": ["333.333.333.333.60:4003"]
  "44.44.44.44": ["444.444.444.444:4004"]


lighthouse:
  am_lighthouse: false
  interval: 60
  hosts:
    - "11.11.11.11"
    - "22.22.22.22"
    - "33.33.33.33"
    - "44.44.44.44"
listen:
  host: 0.0.0.0
  port: 0

punchy:
  punch: true
  respond: true

tun:
  disabled: false
  dev: nebula1
  drop_local_broadcast: false
  drop_multicast: false
  tx_queue: 500
  mtu: 1300
  routes:

logging:
  level: info
  format: text

firewall:
  conntrack:
    tcp_timeout: 12m
    udp_timeout: 3m
    default_timeout: 10m
    max_connections: 100000

  inbound:
    - port: any
      proto: any
      host: any
  outbound:
    - port: any
      proto: icmp
      host: any
```

**Comment by andrewmgee** [comment]
Never mind, figured it out!

Looks like some other app that got installed around the same time I was testing, added the below config to the `/etc/systemd/network/10-allifs.network` file. I had uninstalled all prior apps I thought could have caused the issues, but it didn't modify the file back.

```
[Match]
Name=*

[Network]
DHCP=true
```

I modified the name match to be more specific, so it doesn't interfere with the nebula interface.

Hopefully that helps if someone else has similar issue.
